src/app/core/data/data.service.ts

```diff
@@ -90,4 +101,15 @@ export abstract class DataService<TNormalized extends CacheableObject, TDomain>
     // return this.rdbService.buildSingle(href));
   }
 
+  findEmbedded<TEmbeddedNormalized extends CacheableObject, TEmbeddedDomain>(
```

Comment:
It isn't necessary to write a method to specifically fetch embedded resources in a response. If, for example, you want the collections that were embedded in a community response, all you need to do is first fetch the community, and later do a collectionDataService.findAll scoped to that community. If the collections were embedded in the first response, they will be cached, and no new request will be made.

Reply:
sorry for the late comment. I have reviewed the issue with @atarix83 and unfortunately it is much more complex than what we initially expect.
We can for sure implement the /collections/search/findByCommunity?scope=:uuid and this would probably make possible to fastly solve the issue but inspecting the code I have understood that our angular ui is currently build without taking advantage of the HAL support that we are forcing in the rest part.
In a HAL response you receive a list of links to follow, and the client should be written without previous knowledge about the target URL. In the best scenario the client doesn't need to know anything but of course this lead to very generic client such as the HAL browser. The normal scenario requires that the client (angular-ui) only know about the name of the links to follow, so in this case collections and it should discover the URL of the endpoint looking to the response.
The current services in the angular-ui project have constants were the endpoints are hardcoded, and most danger also "sub-endpoints" are constructed on the angular side.
Also the "main" endpoints can be discovered at runtime just looking to the /api ROOT HAL document.
Making use of the HAL links we will make the angular-ui resilient to endpoints reorganisation but more important we will allow extension or customization on the REST part using different strategies. An institution could provide an additional enhanced endpoint to retrieve the collections without removing the out-of-box implementation but only forcing the HAL document to point to the new endpoints. The angular-ui or some components could be build to inspect the HAL response for not-yet known links and react to them calling external plugins...

I suggest to start implementing a followLink method in the data.service.ts that will accept a
HALdocument and a linkName and will return a HALdocument.